### ISSUE TITLE
Kumar2024 preregistration: prospective calibration frontier and cohort-aware analysis

### ISSUE THREAD

**Issue by sidhulyalkar** [sidhulyalkar/neurOS-v1#27]
## Why preregister this study

Kumar2024 should be the first real longitudinal EEG evidence bundle for neurOS because it asks a deployment-relevant question: **how much target-session calibration is required after learning only from earlier sessions?**

Before running the public study at scale, freeze the analysis plan so dataset-specific choices cannot drift after seeing results.

## Upstream facts to preserve

Current upstream MOABB documents:

- 18 healthy BCI-naive participants;
- 6 sessions on separate days;
- MOABB session `0` is the offline calibration session;
- MOABB sessions `1`-`5` are the five online sessions in chronological order;
- session `0`: 4 bar-feedback runs = 80 trials = 40/class;
- session `1`: 4 online bar runs = 80 trials = 40/class;
- sessions `2`-`5`: 3 online bar runs = 60 trials = 30/class;
- 22 EEG channels, 512 Hz;
- left-hand vs right-hand motor imagery;
- original paper analysis band: 8-30 Hz;
- MOABB excludes car-racing runs and exposes the bar-feedback runs.

The current upstream loader constructs session keys in acquisition order: `0`, then `1` through `5`.

### Original treatment cohorts are not interchangeable

The neural trajectories were collected under two different adaptation/training protocols:

- MOABB subjects **1-9**: Generic Recentering (GR), unsupervised domain adaptation;
- MOABB subjects **10-18**: Personally Assisted Recentering (PAR), supervised recalibration.

This is a meaningful experimental covariate. Do not pool GR and PAR without preserving cohort identity in the evidence artifact and analysis.

## Primary prospective split

Use:

```text
history_policy = prior
```

For every participant:

```text
target session 1: train on 0
session 2:        train on 0,1
session 3:        train on 0,1,2
session 4:        train on 0,1,2,3
session 5:        train on 0,1,2,3,4
```

Future sessions are excluded.

This yields **18 subjects × 5 target sessions = 90 subject-session cases** before any data-quality failures.

Keep `all-other` as a separately labeled sensitivity analysis only. It is not prospective evidence.

## Calibration frontier

Primary labeled calibration budgets:

```text
0, 1, 2, 5, 10 examples / class
```

With a 0.5 fixed evaluation fraction, the nominal dataset structure supports 10/class in every online target session:

- MOABB session `1`: 40/class -> nominal 20/class calibration pool;
- sessions `2`-`5`: 30/class -> nominal 15/class calibration pool.

The strict paired-budget contract remains authoritative: if preprocessing/data issues reduce any actual case below 10/class, fail the promoted run rather than dropping that case from the right side of the curve.

## Evaluation split seeds

Do not choose the calibration/evaluation split after viewing model results.

### Pilot

Use one frozen seed only:

```text
2026
```

The pilot exists to validate data provenance, artifact semantics, runtime and feasibility. It is not the final model-comparison claim.

### Promoted comparison

Use a predeclared small ensemble of calibration/evaluation split seeds, with every competing method sharing each exact split fingerprint. Recommended initial set:

```text
2026, 3407, 9109
```

Report seed-level results and a seed-aggregated subject-session estimate. Do not search over these seeds.

## Initial transparent baseline

Method:

```text
8-30 Hz MOABB trials
-> CSP
-> LDA
```

This is the floor, not the neurOS claim.

Record exact CSP configuration, MNE/sklearn/MOABB versions, fit time and trial inference time.

## Method ladder after the floor is frozen

Under Issue #26, compare the exact same split/calibration identities across:

1. CSP + LDA;
2. EEGNet;
3. EEG-Conformer;
4. frozen representation + matched readout;
5. SourceWeigher transfer on one common frozen representation space;
6. future verified foundation encoders;
7. future ORION EEG representation only after an explicit EEG representation contract exists.

## SourceWeigher authority

For a target session, SourceWeigher may use only:

- embeddings from declared prior source sessions;
- embeddings from the declared target-session **calibration subset** at that budget.

It may not use final evaluation examples to estimate target moments.

At labeled budget `0`, target-dependent SourceWeigher is undefined unless a separate unlabeled-observation budget is explicitly introduced.

## Primary outcome

For method comparison, make the primary endpoint the **paired area under the calibration frontier** over the predeclared budgets `0,1,2,5,10` using balanced accuracy.

Why:

- rewards zero/few-shot usefulness rather than only the largest calibration budget;
- directly measures the recalibration burden neurOS/ORION are intended to reduce;
- uses the whole predeclared frontier;
- is harder to game with one favorable operating point.

Secondary outcomes:

- balanced accuracy at 0/class;
- balanced accuracy at 5/class;
- balanced accuracy at 10/class;
- ROC-AUC;
- Brier score / calibration error where probabilities are valid;
- fit/adaptation wall-clock;
- inference latency;
- failure rate;
- labels required to cross a predeclared performance threshold, once that threshold is chosen independently of the test results.

## Repeated-measures statistics

The 90 subject-session cases are **not 90 independent participants**.

Do not report naive trial/session-level confidence intervals as if every case were independent.

Recommended inference:

- paired method differences at the subject-session level;
- aggregate within subject before participant-level inference, or use a hierarchical/cluster bootstrap;
- preserve GR/PAR cohort as a stratification/covariate;
- report cohort-specific estimates in addition to the pooled estimate;
- report session-index effects because source-history size grows over time;
- for neural models, include predeclared model seeds and separate optimization variance from participant/session variance.

## Cohort-aware reporting

Every result row or linked case manifest should expose:

```text
subject
original_protocol = GR | PAR
held_out_session
source_sessions
history_policy
calibration_per_class
split_seed
partition_fingerprint
calibration_split_fingerprint
```

Primary public plots should offer:

- pooled frontier with cohort-aware uncertainty;
- GR frontier;
- PAR frontier;
- zero-calibration performance by target-session index.

This prevents a decoder improvement from being confused with the different online training histories of the two original experimental groups.

## Pilot execution

After PR #25 merges, first run a **small provenance pilot**, not the full headline study.

Recommended subjects:

```text
1,10
```

This deliberately includes one GR and one PAR participant.

Use:

```text
budgets = 0,1,2,5,10
history_policy = prior
seed = 2026
evaluation_fraction = 0.5
```

Inspect the complete evidence bundle manually before scaling.

Pilot acceptance criteria:

- observed sessions exactly match `0..5`;
- source sessions follow the prospective schedule above;
- cohort identity is preserved;
- all five budgets remain paired across all eligible cases;
- artifact hashes verify;
- no evaluation example enters preprocessing/adaptation fitting;
- runtime is practical enough to scale;
- report and raw CSV agree exactly.

## Scale-up gate

Only after the pilot passes:

1. freeze exact dependency/data identities;
2. run all 18 subjects for the CSP+LDA floor;
3. inspect failures and counts without changing the protocol to improve results;
4. freeze the benchmark bundle/split identities;
5. add EEGNet/EEG-Conformer/SourceWeigher under the same authority;
6. run the predeclared split/model seeds;
7. generate the Evidence Console directly from the artifact bundle.

## Claim boundary

This study can establish offline real-dataset evidence for prospective session transfer and recalibration burden.

It cannot by itself establish:

- live hardware reliability;
- online closed-loop benefit of neurOS;
- clinical benefit;
- ORION superiority unless ORION is actually evaluated under this frozen protocol;
- independence from the original GR/PAR experimental intervention.

**Comment by sidhulyalkar** [comment]
## 2026-08-28 preregistration reconciliation with NSQ PR #87

PR #87 now implements the canonical NSQ composition and a real-data provenance pilot path, but the convenient `full` execution profile currently in that draft must **not** yet be treated as the promoted comparison specified by this issue.

Before the promoted all-subject comparison is allowed to run, #87 or its immediate successor must reconcile these preregistered requirements explicitly:

1. **Split-seed ensemble:** promoted cases use shared split seeds `2026, 3407, 9109`, not only the pilot seed 2026. Every method must consume the exact same case-authority SHA for each split seed.
2. **Primary study endpoint:** the scorecard primary *metric* remains balanced accuracy, but the promoted study primary *endpoint* is paired normalized area under the balanced-accuracy-vs-calibration frontier across `0,1,2,5,10` examples/class.
3. **Paired frontier inference:** method comparisons should compute each participant's complete calibration-frontier AUC first, then perform paired participant-level inference. Per-budget differences remain secondary operating-point analyses.
4. **Optimization variance:** neural methods need predeclared model-seed handling before a headline comparison. The pilot may use one seed strictly for plumbing/feasibility, but a promoted neural comparison must not hide optimization variance.
5. **No protocol repair after inspection:** if real pilot execution shows a scientific protocol defect, fix it before any promoted final-assessment run and version the resulting protocol identity. Do not tune the promoted protocol after seeing all-subject final results.

The one-time real-data CSP pilot added to PR #87 is therefore an execution/provenance test, not a preregistered efficacy result. Its artifact should be inspected for chronology, paired budget support, processed signal geometry, failure preservation, and bundle integrity before these full-study semantics are frozen.

**Comment by sidhulyalkar** [comment]
PR #87 preregistration audit found an important promotion guard to add before the 18-subject run.

The current study code/docs contain a convenience `full_config()` / `--profile full` using all 18 subjects and 20 Braindecode epochs, but that shortcut currently has only the single pilot split seed and does **not** yet encode this issue's promoted split ensemble `2026, 3407, 9109` or a predeclared neural-model seed design. It must therefore not be described or executable as the promoted full comparison.

Proposed gate after the corrected CSP pilot:
1. demote/block the current `--profile full` shortcut rather than let it look publication-ready;
2. represent promoted split seeds explicitly in study identity, not as an execution-loop detail;
3. make normalized paired balanced-accuracy calibration-frontier AUC the primary cross-method endpoint;
4. predeclare neural optimization seeds and report them separately from split/participant/session variation;
5. avoid tuning epoch count after final-assessment results;
6. preserve every split-seed/model-seed failure row;
7. generate one aggregate only from methods sharing exact case/split identities.

We should also design this with compute in mind rather than blindly creating a 3 split x 3 model-seed x 90 case x 5 budget retraining explosion. A source-pretrain / frozen-state or explicitly declared adaptation protocol may be scientifically cleaner and much cheaper, but it must be frozen before final results and applied identically across methods where the comparison claims equivalence.

**Comment by sidhulyalkar** [comment]
### Preregistration refinement before scale-up

PR #87 has now made several implicit choices executable enough that the promoted comparison should tighten this issue before any all-18-subject run.

#### 1. Keep participant as the only independent population unit

The repeated-measures hierarchy for the promoted study should be treated as:

```text
participant
  -> target session
      -> split seed
          -> neural-model seed, when applicable
              -> calibration budget
```

Split seeds and neural-model seeds are algorithmic/repeated-measure perturbations. They do **not** increase the number of independent people. Preserve every seed-level row, but aggregate repeated stochastic measurements within participant before participant-level inference or use a clearly specified participant-cluster/hierarchical procedure.

#### 2. Freeze complete-frontier pairing

The implementation now uses a stricter frontier rule that should be part of the preregistration:

- a subject-session contributes to frontier AUC only when every declared calibration budget has a valid result for that method;
- a participant contributes to a method-level frontier AUC only when every declared target session has a complete subject-session frontier;
- paired method frontier-AUC differences use only participants with complete frontiers for both methods.

This prevents failures at one budget from silently changing the session composition at another budget.

#### 3. Primary study endpoint

Promote the already implemented study-level endpoint name explicitly:

```text
paired_normalized_balanced_accuracy_frontier_auc
```

Balanced accuracy remains the primary pointwise metric. The paired normalized frontier AUC is the primary **method-comparison endpoint** because the scientific question is calibration burden rather than best single-budget accuracy.

#### 4. Add one final classical comparator, then stop baseline creep

Before neural baselines, add the standard Riemannian covariance -> tangent-space -> logistic-regression comparator under the exact same NSQ authority. Current MOABB cross-session/learning-curve examples use this family alongside CSP, so it is a reviewer-relevant omission if we stop at CSP alone.

After CSP+LDA and the Riemannian comparator, do not keep adding classical models without a specific falsification need.

#### 5. Deep-model efficacy requires a separate frozen authority

Direct Braindecode adapters are useful integration contracts but should not yet be interpreted as fair efficacy baselines. Issue #90 now owns the required freeze for:

- deterministic input-unit conversion;
- train-authorized fitted standardization/state;
- validation/checkpoint-selection authority;
- optimizer/training schedule;
- neural-model seeds independent of split seeds;
- environment identity and human-auditable row membership.

The default Kumar2024 pilot is therefore CSP+LDA only. Deep methods require explicit opt-in until #90 is qualified.

#### 6. Environment and row identity

Before the promoted full run, bind a broader execution-environment manifest than the current selected package-version table and expose content-addressed participant/session/run/local-epoch row identities where MOABB metadata permits. The existing processed-data SHA remains the machine authority; this addition improves reviewer auditability.

These refinements do not change the pilot result. They prevent the scale-up from turning stochastic repeats, failures, or baseline-specific preprocessing choices into hidden degrees of freedom.

**Comment by sidhulyalkar** [comment]
Preregistration execution gate update:

The promoted comparison should now treat #93 (**NSQ Study Materialization Authority**) as a prerequisite to any new neural/all-subject final-assessment run.

This does not change the frozen scientific endpoints, budgets, prospective history policy, or shared split seeds in this issue. It strengthens the provenance needed to execute them credibly.

Before promoted execution, #93 should freeze and bind:

1. exact realized environment authority;
2. byte-addressed raw Kumar2024 materialization;
3. human-auditable participant/session/run/local-epoch observation identities and role membership;
4. a composed study materialization identity carried into the NSQ evidence bundle.

PR #92 separately freezes EEGNet optimization/validation/state-selection authority. Once both #92 and #93 are qualified, the remaining preregistration work is to finalize the multi-split / model-seed analysis hierarchy before inspecting all-subject neural results.

No result-driven protocol repair is authorized by this change.

**Comment by sidhulyalkar** [comment]
### Promoted comparison authority is now durable

PR #97 has merged to `main` as `a3f2d474b77acbf88042cb4c2b80920f2b7d34ed`, preserving exact qualified head `8b422009ef7e66804f639ef2f02a36cf23bff120` as an ancestor.

The authority now durably freezes:

- split seeds `2026, 3407, 9109`;
- EEGNet optimization seeds `31415, 384165836, 3991196546`;
- participant as the only independent population unit;
- participant -> session -> split seed -> model seed -> calibration budget aggregation;
- strict complete-frontier pairing;
- primary endpoint `paired_normalized_balanced_accuracy_frontier_auc`;
- failure-preserving stochastic diagnostics that do not inflate N;
- exact case-authority equality across competing methods/model seeds.

The additional predeclared training-only feasibility gate also passed before merge. The slowest real-Kumar bracket EEGNet fit was 7.55 minutes on the GitHub-hosted CPU reference environment, below the predeclared 15-minute keep-plan threshold. No final-assessment prediction or efficacy metric was produced by that probe, so the full 3x3 stochastic design remains unchanged.

The temporary feasibility branch has been reset to the durable main merge, so its one-shot workflow/script are no longer present on that branch. The historical run/artifact remain addressable for audit.

I am keeping this issue open until the exact-main post-merge authority checks are green and the promoted execution-plan contract binds this comparison-plan SHA into sharded result artifacts. The next step is orchestration, not ORION and not an efficacy launch.
